internal/database/repository/queue.go

```diff
@@ -51,10 +53,14 @@ func (s *PostgresRepository) UpsertQueues(ctx context.Context, queues []*dao.Par
 		is_managed = EXCLUDED.is_managed,
 		max_running_apps = EXCLUDED.max_running_apps,
 		running_apps = EXCLUDED.running_apps`
+
 	for _, q := range queues {
-		parentId, err := s.getQueueID(ctx, q.Parent, q.Partition)
-		if err != nil {
-			return fmt.Errorf("could not get parent queue from DB: %v", err)
+		var err error
```

Comment:
Is the expected number of queues always going to be small? Asking for a friend who thinks this will query N times per upsert.

Reply:
@dave-gantenbein 
Hmm... we might need to think this through a bit more. Given that the queues are organized in a parent-child structure and can be nested down to `D` levels, we need to insert the parent first to allow for inserting its children. While we could potentially insert all the direct children in a single query, in that case, time complexity will be `O(D)`, where `D` is the depth of the tree. This will be a slight improvement but the worst case will still be `O(N)` where `N==D` for a skewed tree
`N = Number of nodes in the tree`

Reply:
We could bundle insertions of child queues into single INSERT transactions (perhaps all siblings at each generation), and then downward - or all descendants of the top-level queue in one INSERT (or txn). But I'd be surprised if an organization would be creating more than hundreds of queues per day - I expect `applications`  will get the most traffic each day, by far. 

Note that in the changed code (lines 59-63), we call `getQueueID(..)` only if `UpsertQueues(..)` is not passed a given parentID - so the first/top-level call to `UpsertQueues()` will cause this to query for a queue ID, then all child queues will have this queue's id as its parentID, and so on down the descendants tree. There should be only one call to getQueueID() in the chain.